main.py

```
@@ -113,5 +113,160 @@ async def list_esp_serial_ports() -> (str, str):
+    else:
+        erase_cmd = f"bash -c 'source {export_script} && idf.py erase-flash'"
+    
+    returncode, stdout, stderr = await run_command_async(erase_cmd)
+    
+    open('mcp-erase.log', 'w+').write(str((stdout, stderr)))
+    logging.warning(f"erase-flash result {stdout} {stderr}")
+    return stdout, stderr
+
+
+@mcp.tool()
+async def monitor_esp_device(project_path: str, port: str = None, timeout_seconds: int = 30) -> (str, str):
```

Ruff (0.14.14)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)